File: output/llama3.1-8b/avatar/Python/Java/codeforces_287_A.java
import java.util.Scanner;

public class codeforces_287_A {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int[][] t = new int[4][4];

        for (int i = 0; i < 4; i++) {
            String input = scanner.nextLine();
            for (int j = 0; j < 4; j++) {
                if (input.charAt(j) == '#') {
                    t[i][j] = 1;
                } else {
                    t[i][j] = 0;
                }
            }
        }

        int black = 0;
        int white = 0;

        for (int i = 0; i < 3; i++) {
            for (int j = 0; j < 3; j++) {
                if (t[i][j] == 1) {
                    black++;
                } else {
                    white++;
                }
                if (i < 3 && t[i + 1][j] == 1) {
                    black++;
                } else {
                    white++;
                }
                if (j < 3 && t[i][j + 1] == 1) {
                    black++;
                } else {
                    white++;
                }
                if ((i < 3 && j < 3) && t[i + 1][j + 1] == 1) {
                    black++;
                } else {
                    white++;
                }

                if (black == 0 || black == 3 || white == 0 || white == 3 || white == 4 || black == 4) {
                    System.out.println("YES");
                    return;
                }
                black = 0;
                white = 0;
            }
        }
        System.out.println("NO");
    }
}

File: output/llama3.1-8b/avatar/Java/Python/codeforces_287_A.py
import sys

def main():
    str_list = []
    for _ in range(4):
        str_list.append(input())

    for i in range(2):
        for j in range(2):
            if (str_list[i][j] == str_list[i][j+1] and 
                (str_list[i+1][j] == str_list[i][j] or str_list[i+1][j+1] == str_list[i][j])) or \
               (str_list[i+1][j] == str_list[i+1][j+1] and 
                (str_list[i][j] == str_list[i+1][j] or str_list[i][j+1] == str_list[i+1][j])):
                print("YES")
                return

    print("NO")

if __name__ == "__main__":
    main()